visualization/motion_planning_rviz_plugin/src/motion_planning_frame_planning.cpp

```diff
@@ -129,15 +135,31 @@ void MotionPlanningFrame::computePlanButtonClicked()
 void MotionPlanningFrame::computeExecuteButtonClicked()
 {
   if (move_group_ && current_plan_)
-    move_group_->execute(*current_plan_);
+    ui_->stop_button->setEnabled(true);
+    move_group_->asyncExecute(*current_plan_);
+}
+
+void MotionPlanningFrame::computeStopButtonClicked()
+{
+  if (move_group_) {
+    current_plan_.reset();
+    move_group_->stop();
+    move_group_->clearPoseTargets();
+    move_group_->setJointValueTarget(move_group_->getCurrentJointValues());
+    move_group_->move();
```

Comment:
Why is stop() not enough? Does it not cancel the ActionServerRequest which should stop execution?